scripts/pre_commit_linter.py

```diff
@@ -868,17 +870,19 @@ def _check_newline_character(all_files):
         filename for filename in all_files if not
         any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
 
-    for filename in all_files:
-        content = FileCache.read(filename, mode='rb')
-        files_checked += 1
-        if len(content) == 1:
-            errors_found += 1
-            print '%s --> Error: Only one character in file.' % filename
-        elif len(content) >= 2 and not re.match(r'[^\n]\n', content[-2:]):
-            errors_found += 1
-            print (
-                '%s --> Please ensure that this file ends with exactly one '
-                'newline char.' % filename)
+    with _redirect_stdout(_TARGET_STDOUT):
+        for filename in all_files:
+            content = FileCache.read(filename, mode='rb')
+            files_checked += 1
+            if len(content) == 1:
+                errors_found += 1
+                print '%s --> Error: Only one character in file.' % filename
+            elif len(content) >= 2 and not re.match(r'[^\n]\n', content[-2:]):
+                errors_found += 1
+                print (
+                    '%s --> Please ensure that this file ends with exactly one '
+                    'newline char.' % filename)
+                print ''
```

Comment:
Why do we need this?

Reply:
This extra line is added just to make things look prettier in the summary of errors section. I am adding screenshots of the summary of error section with and without the above extra line. Please check it.